sdk/resourcemanager/azure-resourcemanager-authorization/src/main/java/com/azure/resourcemanager/authorization/implementation/RoleAssignmentsImpl.java

```diff
@@ -78,7 +80,7 @@ public RoleAssignmentImpl getByScope(String scope, String name) {
     @Override
     public PagedFlux<RoleAssignment> listByServicePrincipalAsync(String principalId) {
         String filterStr = String.format("principalId eq '%s'", Objects.requireNonNull(principalId));
-        return PagedConverter.mapPage(inner().listAsync(URLEncoder.encode(filterStr, StandardCharsets.UTF_8), null), this::wrapModel);
+        return PagedConverter.mapPage(inner().listAsync(urlEncode(filterStr), null), this::wrapModel);
```

Comment:
Nice catch.

I thought build under Java 8 will fail this at compile, no idea why CI passed before this...